### DISCUSSION TITLE
flood prevention

### DISCUSSION THREAD

**Discussion by DalisonCPU** [BearWare/TeamTalk5#1968] [Ideas]
### Description

Many people are using scripts to connect to TeamTalk servers to flood users and this is becoming more and more frequent. The TeamTalk flood system does not work as it should, because even putting 5 commands in 3000 MS a user barely entered the server manages to send more than 20 30 messages to a user via private message.
To solve this problem, I suggest implementing a functionality that allows defining the time that a user can interact with other users after entering the server. This option could be set on users, and could be chosen after how many seconds a user can send text and voice messages.

### Application

- [ ] qtTeamTalk
- [ ] TeamTalkAndroid
- [ ] iTeamTalk
- [X] TeamTalkServer

**Comment by bear101** [comment]
The best way to avoid abuse on a server is to use BearWare.dk WebLogin.

**Reply by Ed-Fe** [reply]
I do not agree. I think other solutions should also be implemented.

**Comment by DalisonCPU** [comment]
If you want this system to be really used, create a more practical way to create an account using this system since not even 20% of TeamTalk users use the login system. And for you it is not difficult to implement this system. Understand, people will not be wasting time to create an account on a website just to log into a server, and you're forgetting that most are visually impaired, which makes this more difficult. With 3 minutes of your time you can implement this. Many people will thank you for this. If you have any doubts, I can prove it to you. Make a survey and make it available and see the responses.

**Reply by Nardol** [reply]
> If you want this system to be really used, create a more practical way to create an account using this system since not even 20% of TeamTalk users use the login system. And for you it is not difficult to implement this system. Understand, people will not be wasting time to create an account on a website just to log into a server, and you're forgetting that most are visually impaired, which makes this more difficult. With 3 minutes of your time you can implement this. Many people will thank you for this. If you have any doubts, I can prove it to you. Make a survey and make it available and see the responses.

The fact you know it is not difficult and can estimate the time to implement it suggests you know how to do it so why don't you make a pull request "With 3 minutes of your time"?
I am sure many people will thank you for this.

It also looks like you have very precise account of visually impaired TeamTalk users and percentage who use Bearware account. I would really be interested in knowing your official sources... Or is it only a troll as it seams?

As a last point and as a blind user myself, I really don't see where is the difficulty to create a Bearware account using screen reader. It took about three minutes of my time to do it. This kind of comments is one of the reasons why accessibility issues are not taken seriously by a lot of developers. And I say it as a developer implied in accessibilyty myself.
I am not @bear101 and I sometimes disagree with some of his reactions but what you say is counterproductive IMHO. You are frustrated which I can fully understand but that's all.

**Reply by bear101** [reply]
Hear, hear! @Nardol

**Reply by DalisonCPU** [reply]
I'm not a fluent C++ programmer but I know a little and I have an idea of ​​how long it would take me to create my suggestion. I think that if you don't have anything to contribute to the post, you also don't need to comment just to increase the developer's hego.

**Comment by muamel1** [comment]
Many people complain about not receiving an email containing a code or activation link when creating an account on bearware.dk.

**Comment by muamel1** [comment]
If TeamTalk wants to compete with existing programs, it must keep up with the functionality of other applications and come up with creative ideas that give it a competitive edge.
In reality, I see TeamTalk as much better than Claphouse. However, the reason for its lack of widespread adoption is its deficiency in these essential features.
Few programs require you to visit their website and create an account. The majority of applications allow you to create an account through a simple user interface within the app, and I believe @bear101  can do the same.
I agree that @DalisonCPU may have exaggerated by suggesting it could be done in a few minutes, but that doesn't necessarily mean the suggestion should be completely dismissed.

**Reply by Nardol** [reply]
> If TeamTalk wants to compete with existing programs

But is it really the goal of TeamTalk? Only @bear101 can answer.

If someone makes a pull request, I don't see why it would be simply rejected.
It looks simply not a todo because it looks like there are other priorities so the workaround is to use bearware.dk weblogins. Unfortunately, scripts does not use weblogin I suppose.

I also think an integrated interface to create a bearware.dk account could be a good feature, but it looks a little off-topic.
Once again, a pull request could be made by someone who can make the GUI and communicate with APIs to create the account. Why not a separate discussion with constructive arguments before? As this one looks to be about flood prevention.

**Comment by DalisonCPU** [comment]
It's just a suggestion. If you think that 5 minutes of your time to help hundreds of users will hurt you or be bad, that's your decision. I'm honestly giving up sending suggestions to TeamTalk, because I've already sent many and what you do is look for reasons not to implement them. Honestly, I don't know why you have this guide, since 99.5% of what is proposed here is completely ignored and no, it's not because it's bad... It's because you don't want to implement it. If you want to hear suggestions, really listen, pretending you're interested in what users think isn't enough. Of course, if you want to end this thread by saying that I was extremely arrogant with you, that's within your rights.
And one more thing. Why doesn't it make sense to implement this, if even Discord has this option and Discord is one if not the biggest voice chat program in the world for games? I dare you to show 10 servers with bearware.dk web login. If he was that good, why don't most people use them? I leave the reflection here.

**Comment by beqabeqa473** [comment]
If you think that it is five minutes of work, just sit and do it
yourself and send a pr. I don't think that author won't accept your pr
if this feature will be usable for lots of people.
Why you don't want to dedicate these five minutes as you said?


On 8/2/23, DalisonCPU ***@***.***> wrote:
> It's just a suggestion. If you think that 5 minutes of your time to help
> hundreds of users will hurt you or be bad, that's your decision. I'm
> honestly giving up sending suggestions to TeamTalk, because I've already
> sent many and what you do is look for reasons not to implement them.
> Honestly, I don't know why you have this guide, since 99.5% of what is
> proposed here is completely ignored and no, it's not because it's bad...
> It's because you don't want to implement it. If you want to hear
> suggestions, really listen, pretending you're interested in what users think
> isn't enough. Of course, if you want to end this thread by saying that I was
> extremely arrogant with you, that's within your rights.
> And one more thing. Why doesn't it make sense to implement this, if even
> Discord has this option and Discord is one if not the biggest voice chat
> program in the world for games? I dare you to show 10 servers with
> bearware.dk web login. If he was that good, why don't most people use them?
> I leave the reflection here.
>
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/BearWare/TeamTalk5/discussions/1968#discussioncomment-6610819
> You are receiving this because you are subscribed to this thread.
>
> Message ID:
> ***@***.***>


-- 
with best regards Beqa Gozalishvili
Tell: +995593454005
Email: ***@***.***
Web: https://gozaltech.org
Skype: beqabeqa473
Telegram: https://t.me/gozaltech
facebook: https://facebook.com/gozaltech
twitter: https://twitter.com/beqabeqa473
Instagram: https://instagram.com/beqa.gozalishvili

**Reply by DalisonCPU** [reply]
As I said, I'm not a fluent C++ developer to implement this feature... But yeah, I think it would take about five minutes. Maybe I got it a little wrong, but if the suggestion is good you could have done it. Another thing. Did you just respond to the part that hurt you? Where are the 10 bearware.dk servers? Why is it not a good option if Discord uses it?
